documentation/faq/70-packages.md

```diff
@@ -12,4 +12,6 @@ Old beta versions of the SvelteKit template included the configuration value `no
 
 The second most commonly-encountered issue is having a Svelte component that imports a CommonJS library. In this case, you should try to work with the library authors to distribute an ESM version of the dependency. However, in the meantime, you can workaround this issue by adding the dependency to `vite.optimizeDeps.include` in `svelte.config.js`.
 
-Finally, Vite has had some issues that have been fixed, so we recommend upgrading to the latest version of Vite. If you are still encountering issues we recommend searching both [the Vite issue tracker](https://github.com/vitejs/vite/issues) and the issue tracker of the library in question. Sometimes issues can be worked around by fiddling with the [`optimizeDeps`](https://vitejs.dev/config/#dep-optimization-options) or [`ssr`](https://vitejs.dev/config/#ssr-options) config values.
+Some issues may also be specific to Vite, so we recommend searching both [the Vite issue tracker](https://github.com/vitejs/vite/issues) and the issue tracker of the library in question. Sometimes issues can be worked around by fiddling with the [`optimizeDeps`](https://vitejs.dev/config/#dep-optimization-options) or [`ssr`](https://vitejs.dev/config/#ssr-options) config values. 
+
+Finally, do check out `@sveltejs/vite-plugin-svelte`'s docs on the [limitation of importing some Svelte libraries](https://github.com/sveltejs/vite-plugin-svelte/tree/main/packages/vite-plugin-svelte#importing-third-party-svelte-libraries).
```

Comment:
We avoid bundling Svelte components automatically https://github.com/sveltejs/kit/pull/1148. I wonder if we should also automatically avoid optimizing them as well?

Reply:
There's a [discussion in discord](https://discord.com/channels/457912077277855764/819723698415599626/850050004538228766) from @dominikg regarding auto excluding optimizations of Svelte libraries. Looks like the exact heuristics are already implemented in that PR ([this code specifically](https://github.com/sveltejs/kit/blob/17bf30143c14d3b45f7d116487bfd74f9e271236/packages/kit/src/core/utils.js#L106-L108)), but only for `ssr.noExternals`.

I think the specific logic could live in vite-plugin-svelte instead, though I'm not sure if @dominikg has plans for that, or to fix the underlying prebundling issue directly.

Reply:
FYI I've dug into the prebundling issue last week and wrote my findings [here](https://github.com/bluwy/vite-svelte-dedupe#readme). It should explain the reason why `optimizeDeps.exclude` work as it is for Svelte.

@dominikg had raised some concerns in discord regarding the auto-exclude feature being directly in vite-plugin-svelte. I'll let him explain further, but perhaps we could handle it in SvelteKit for now.

Reply:
Wow, that's some great investigation. I have to be honest I'm not totally following everything you've written as my understanding of the Vite internals is minimal. @dominikg, @GrygrFlzr, and @Rich-Harris probably know more about it than I do. It would be nice to file an issue that we can link to here or on the `vite-plugin-svelte` readme that explain the reason for this restriction and so that people can follow along in case it is resolved at some point

Reply:
@benmccann Sorry for the long follow-up :sweat_smile: I've finally opened an issue at https://github.com/vitejs/vite/issues/3910. Will start linking the issue to kit or vite-plugin-svelte if it proves to be a hard problem to tackle in Vite side.

Reply:
I sent a PR for that here: https://github.com/sveltejs/vite-plugin-svelte/pull/69